src/PostRector/NameImportingPostRector.php

```diff
@@ -1,219 +0,0 @@
-        $this->phpDocInfoFactory = $phpDocInfoFactory;
-        $this->nodeNameResolver = $nodeNameResolver;
-    }
-
-    public function enterNode(Node $node): ?Node
-    {
-        $autoImportNames = (bool) $this->parameterProvider->provideParameter(Typo3Option::AUTO_IMPORT_NAMES);
-        if (! $autoImportNames) {
-            return null;
-        }
-
-        if ($this->shouldSkip($this)) {
-            return null;
-        }
-
-        if ($node instanceof Name) {
-            return $this->processNodeName($node);
-        }
-
-        $importDocBlocks = (bool) $this->parameterProvider->provideParameter(Option::IMPORT_DOC_BLOCKS);
-        if (! $importDocBlocks) {
-            return null;
-        }
-
-        $phpDocInfo = $this->phpDocInfoFactory->createFromNodeOrEmpty($node);
-        $this->docBlockNameImporter->importNames($phpDocInfo, $node);
-
-        return $node;
-    }
-
-    public function getPriority(): int
-    {
-        // The \Rector\PostRector\Rector\NameImportingPostRector::class from Rector itself uses 600, so we go one level up
-        return 601;
-    }
-
-    public function getRuleDefinition(): RuleDefinition
-    {
-        return new RuleDefinition(
-            'Imports fully qualified class names in parameter types, return types, extended classes, implemented, interfaces and even docblocks',
-            [
-                new CodeSample(
-                    <<<'CODE_SAMPLE'
-$someClass = new \Some\FullyQualified\SomeClass();
-CODE_SAMPLE
-                    ,
-                    <<<'CODE_SAMPLE'
-use Some\FullyQualified\SomeClass;
-
-$someClass = new SomeClass();
-CODE_SAMPLE
-                ),
-            ]
-        );
-    }
-
-    private function shouldSkip(PostRectorInterface $postRector): bool
-    {
-        $smartFileInfo = $this->currentFileInfoProvider->getSmartFileInfo();
-
-        if (! $smartFileInfo instanceof SmartFileInfo) {
-            return false;
-        }
-
-        $skip = $this->parameterProvider->provideArrayParameter(Option::SKIP);
-        if ([] === $skip) {
-            return false;
-        }
-
-        $rectorClass = get_class($postRector);
-        if (! array_key_exists($rectorClass, $skip)) {
-            return false;
-        }
-
-        $locations = $skip[$rectorClass];
-        $filePathName = $smartFileInfo->getPathName();
-        if (in_array($filePathName, $locations, true)) {
-            return true;
-        }
-
-        $fileName = $smartFileInfo->getFileName();
-        foreach ($locations as $location) {
-            $ignoredPath = $this->normalizeForFnmatch($location);
-
-            if ($smartFileInfo->endsWith($ignoredPath) || $smartFileInfo->doesFnmatch($ignoredPath)) {
-                return true;
-            }
-
-            if (rtrim($ignoredPath, '\/') . DIRECTORY_SEPARATOR . $fileName === $filePathName) {
-                return true;
-            }
-        }
-
-        return false;
-    }
```

Comment:
This is now covered in Rector itself: https://github.com/rectorphp/rector/pull/5499